provider/spark.go

```diff
@@ -626,8 +630,170 @@ func (e EMRExecutor) InitializeExecutor(store FileStore) error {
 	return store.Write(sparkScriptPath, buff)
 }
 
-func NewSparkExecutor(execType SparkExecutorType, config SparkExecutorConfig, logger *zap.SugaredLogger) (SparkExecutor, error) {
+type SparkGenericConfig struct {
+	Master        string
+	DeployMode    string
+	PythonVersion string
+}
+
+func (sc *SparkGenericConfig) Deserialize(config SerializedConfig) error {
+	err := json.Unmarshal(config, sc)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (sc *SparkGenericConfig) Serialize() ([]byte, error) {
```

Comment:
@sdreyer, I'm seeing a variance in our config struct's `Serialize` method, should we standardize these? For example:
* `PostgresConfig.Serialize` returns `[]byte` only because it panics if it encounters an error
* `RedisConfig.Serialize` returns `SerializedConfig` (type alias for `[]byte`) and also panics if it encounters an error
* `DatabricksConfig.Serialize` returns `([]byte, error)`

The other configs fall into one of these 3 buckets. We should standardize around 1, yeah?

Reply:
P.S. @ahmadnazeri, I'm not proposing that this should fall within the scope of your PR, just wanted to point this out.

Reply:
We should probably remove the panics, but out of scope for now